.github/workflows/ci.yaml

```diff
@@ -0,0 +1,120 @@
+name: Zipline CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-and-test:
+    name: build (${{ matrix.os }}, ${{ matrix.python-version }}, ${{ matrix.pandas-version }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]  # TODO: Add windows-latest, macos-latest
+        python-version: [2.7, 3.5]
+        pandas-version: [0.18.1, 0.22.0]
+        include:  # TODO: Add 3.7, 3.8
+          - python-version: 2.7  # py2.7, old pandas
+            pandas-version: 0.18.1
+            numpy-version: 1.11.3
+            scipy-version: 0.17.1
+          - python-version: 3.5  # py3.5, new pandas
+            pandas-version: 0.18.1
+            numpy-version: 1.11.3
+            scipy-version: 0.17.1
+          - python-version: 3.5  # py3.5, new pandas
+            pandas-version: 0.22.0
+            numpy-version: 1.14.1
+            scipy-version: 1.0.0
+        exclude:
+          - python-version: 2.7
+            pandas-version: 0.22.0
+    env:
+      PYTHON_VERSION: ${{ matrix.python-version }}
+      PANDAS_VERSION: ${{ matrix.pandas-version }}
+      NUMPY_VERSION: ${{ matrix.numpy-version }}
+      SCIPY_VERSION: ${{ matrix.scipy-version }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Pip cache
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/.pip/pip_np$NUMPY_VERSION
+          key: ${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.pandas-version }}-${{ hashFiles('**/requirements.txt') }}-pip-cache
+      - name: Install dependencies
+        run: |
+          if [ "$PANDAS_VERSION" = 0.22.0 ]; then
+            PANDAS_DATAREADER_VERSION=0.4.0 DASK_VERSION=0.17.1
+          fi
+          source ./ci/actions/overwrite_requirements.sh
+          cat etc/requirements.txt
+          TALIB_VERSION=$(cat ./etc/requirements.txt | grep "ta-lib" | sed "s/ta-lib==\([^ ]*\) *.*/\1/")
+          CERTIFI_VERSION=$(cat ./etc/requirements.txt | grep "certifi" | sed "s/certifi==\([^ ]*\) *.*/\1/")
+
+          if [[ "$RUNNER_OS" == "Linux" ]]; then
+              sudo apt-get install python3-venv
+          fi
+          python3 -m venv testenv
+          source ./testenv/bin/activate
+          conda install -c anaconda libgfortran=3.0
+          pip install pip cython numpy pandas scipy ta-lib certifi -c etc/requirements.txt --cache-dir="$HOME/.cache/.pip/pip_np$NUMPY_VERSION"
+          etc/dev-install --cache-dir="$HOME/.cache/.pip/pip_np$NUMPY_VERSION"
+          pip freeze | sort
+          echo "::set-env name=PATH::$PATH"
```

Comment:
This seems like a lot more complexity than I would have expected. A couple quick suggestions:

1. Given that we're using the `setup-python` action to make a fresh python for each of these builds, I don't think it should be necessary to create a separate virtualenv for the build.
2. I don't think we should need to use conda for anything here anymore. In particular, installing `anaconda` brings in a **lot** of stuff we don't want (anaconda is python + all the packages that anaconda the company curates, c.f. https://stackoverflow.com/questions/30034840/what-are-the-differences-between-conda-and-anaconda).
3. I don't think we want to be pre-installing all the packages you currently have listed in the top-level `pip install`. In particular, we probably **don't** want to be installing `scipy` and `pandas` at the same time as we install `numpy`, since the reason we pre-install numpy and Cython is so that those packages are available at build time for scipy and pandas (and other packages that depend on them at build time). I'd expect that the right sequence of installs here should look something like: `pip install pip==<pinned pip version> setuptools=<pinned setuptools version>`, `pip install -r etc/requirements_build.in -c etc/requirements.txt`, `pip install -e .[all] -c etc/requirements.txt`.

Reply:
@ssanderson Sorry you caught this PR in a wip state but appreciate the feedback!
I set up a virtualenv to get around this error that I was running into when I tried to pip install directly to the actions host.
```ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/cython.py'
Consider using the `--user` option or check the permissions.```

I tried to strip away the conda installs and as you warned me last week, pip doesn't handle the dependencies as well  and I was running into a bunch of dependency constraint errors.